frontend/src/components/GeyserStakeView.tsx

```diff
@@ -0,0 +1,42 @@
+import { Wallet } from 'ethers';
+import React, { useContext, useState } from 'react';
+import { MOCK_ERC_20_ADDRESS } from '../constants';
+import { GeyserContext } from '../context/GeyserContext';
+import { VaultContext } from '../context/VaultContext';
+import Web3Context from '../context/Web3Context';
+import { approveCreateDepositStake, approveDepositStake, parseUnitsWithDecimals, permitDepositStake } from '../sdk';
+import { Input } from '../styling/styles';
+
+interface GeyserStakeViewProps {}
+
+export const GeyserStakeView: React.FC<GeyserStakeViewProps> = () => {
+  const [amount, setAmount] = useState<string>('')
+  const { selectedGeyser } = useContext(GeyserContext)
+  const { signer } = useContext(Web3Context)
+  const { selectedVault } = useContext(VaultContext)
+
+  const handleStake = async () => {
+    if (selectedGeyser) {
+      const parsedAmount = await parseUnitsWithDecimals(amount, MOCK_ERC_20_ADDRESS, signer!)
+      const geyserAddress = selectedGeyser.id
+      if (selectedVault) {
```

Comment:
can simplify this to `if(selectedGeyser && selectedVault)`

Reply:
Just to be sure, the `if` is nested because there is two levels of checks:

```
if (selectedGeyser) {
  ...
  if (selectedVault) {
    approveCreateDepositStake
  } else {
    approveDepositStake
  }
}
```

I could change it to something like this, if that's clearer?

```
if (selectedGeyser && selectedVault) {
  ...
  approveCreateDepositStake
} else if (selectedGeyser) {
  ...
  approveDepositStake
}
```

Reply:
Ahh whoops , totally missed that! I think the existing code communicates that better! No changes required